llvm/tools/llvm-foreach/llvm-foreach.cpp

```diff
@@ -222,7 +229,12 @@ int main(int argc, char **argv) {
     }
 
     std::string ErrMsg;
-    // TODO: Add possibility to execute commands in parallel.
+    if (ExecuteInParallel) {
```

Comment:
I suggest that we introduce an option `-j` (similar to one from `make` or `ninja`), which controls the maximum amount of parallel jobs that could be launched. The code then could be generalized, i.e. we always call `ExecuteNoWait`, but once `CommandsStarted` contains maximum amount of jobs, we wait for at least one of them before submitting the next one.

Default value for the new option could be one to be on the safe side

Reply:
Thanks, it's a good idea, I'd like to implement this too. 
But it seems a bigger scope of work, so I'd suggest enabling this option in a separate PR (if I won't find a way to introduce `-j` mode within a few days)